jupyter/main.ipynb

```diff
@@ -169,22 +274,33 @@
     "from dotenv import load_dotenv\n",
     "load_dotenv(override=True)\n",
     "\n",
-    "api_key = \"\" # add your Mistral API key here if needed\n",
+    "api_key = \"\" # Initializes the API key variable.\n",
     "if api_key == \"\":\n",
-    "    api_key = os.getenv(\"MISTRAL_API_KEY\")\n",
+    "    api_key = os.getenv(\"MISTRAL_API_KEY\") # Retrieves the API key from environment variables if not already set.\n",
     "elif not api_key:\n",
```

Comment:
Error handling doesn't work as intended; it didn't tell me when I tried to run it without API key.